mm2src/coins/tendermint/ibc/transfer_v1.rs

```diff
@@ -0,0 +1,107 @@
+use super::{ibc_proto::IBCTransferV1Proto, IBC_OUT_SOURCE_PORT, IBC_OUT_TIMEOUT_IN_NANOS};
+use crate::tendermint::type_urls::IBC_TRANSFER_TYPE_URL;
+use common::number_type_casting::SafeTypeCastingNumbers;
+use cosmrs::{tx::{Msg, MsgProto},
+             AccountId, Coin, ErrorReport};
+use std::convert::TryFrom;
+
+#[derive(Clone, Debug, Eq, PartialEq)]
+pub(crate) struct MsgTransfer {
+    /// the port on which the packet will be sent
+    pub(crate) source_port: String,
+    /// the channel by which the packet will be sent
+    pub(crate) source_channel: String,
+    /// the tokens to be transferred
+    pub(crate) token: Coin,
+    /// the sender address
+    pub(crate) sender: AccountId,
+    /// the recipient address on the destination chain
+    pub(crate) receiver: AccountId,
+    /// Timeout height relative to the current block height.
+    /// The timeout is disabled when set to 0.
+    pub(crate) timeout_height: Option<cosmrs::tendermint::block::Height>,
+    /// Timeout timestamp in absolute nanoseconds since unix epoch.
+    /// The timeout is disabled when set to 0.
+    pub(crate) timeout_timestamp: u64,
+    // Not supported by some of the cosmos chains like IRIS
+    // pub(crate) memo: Option<String>,
+}
+
+impl MsgTransfer {
+    pub(crate) fn new_with_default_timeout(
+        source_channel: String,
+        sender: AccountId,
+        receiver: AccountId,
+        token: Coin,
+    ) -> Self {
+        let timestamp_as_nanos: u64 = common::get_local_duration_since_epoch()
+            .expect("get_local_duration_since_epoch shouldn't fail")
+            .as_nanos()
+            .into_or_max();
+
+        Self {
+            source_port: IBC_OUT_SOURCE_PORT.to_owned(),
+            source_channel,
+            sender,
+            receiver,
+            token,
+            timeout_height: None,
+            timeout_timestamp: timestamp_as_nanos + IBC_OUT_TIMEOUT_IN_NANOS,
+            // memo: Some(memo.clone()),
+        }
+    }
+}
+
+impl Msg for MsgTransfer {
+    type Proto = IBCTransferV1Proto;
+}
+
+impl TryFrom<IBCTransferV1Proto> for MsgTransfer {
+    type Error = ErrorReport;
+
+    fn try_from(proto: IBCTransferV1Proto) -> Result<MsgTransfer, Self::Error> { MsgTransfer::try_from(&proto) }
```

Comment:
sure I can, just for the note, if we don't apply `inline(always)` on non-public functions, it will be useless. just `inline` won't do any difference on non-public functions

Reply:
Thanks for the note, didn't know that :)